src/GLideNHQ/TxHiResCache.cpp

```diff
@@ -281,13 +283,39 @@ TxHiResCache::LoadResult TxHiResCache::loadHiResTextures(const wchar_t * dir_pat
 		pfname = strstr(fname, ident.c_str());
 		if (pfname != fname) pfname = 0;
 		if (pfname) {
-			if (sscanf(pfname + ident.size(), "#%08X#%01X#%01X#%08X", &chksum, &fmt, &siz, &palchksum) == 4)
-				pfname += (ident.size() + CRCFMTSIZ_LEN + PALCRC_LEN);
-			else if (sscanf(pfname + ident.size(), "#%08X#%01X#%01X", &chksum, &fmt, &siz) == 3)
-				pfname += (ident.size() + CRCFMTSIZ_LEN);
-			else
+			uint32_t length = 0;
+
+			if (sscanf(pfname + ident.size(), "#%08X#%01X#%01X#%08X", &chksum, &fmt, &siz, &palchksum) == 4) {
+				length = (ident.size() + CRCFMTSIZ_LEN + PALCRC_LEN);
+			} else if (sscanf(pfname + ident.size(), "#%08X#%01X#%01X", &chksum, &fmt, &siz) == 3) {
+				length = (ident.size() + CRCFMTSIZ_LEN);
+			}
+
+			/* sscanf seems to not care about things that come after the input
+			 * so to combat sscanf's behavior, check if there's '$' in the str
+			 * and re-try sscanf,
+			 * this is needed for wildcard support
+			 */
+			if (strchr(pfname + ident.size(), '$')) {
```

Comment:
Since we do this check anyway, it is better to put the code above in the else clause:
```
if (strchr(pfname + ident.size(), '$')) {
...
} else {
	if (sscanf(pfname + ident.size(), "#%08X#%01X#%01X#%08X", &chksum, &fmt, &siz, &palchksum) == 4) {
		length = (ident.size() + CRCFMTSIZ_LEN + PALCRC_LEN);
	} else if (sscanf(pfname + ident.size(), "#%08X#%01X#%01X", &chksum, &fmt, &siz) == 3) {
		length = (ident.size() + CRCFMTSIZ_LEN);
	}
}

```